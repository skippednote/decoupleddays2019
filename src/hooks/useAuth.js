import { useState, useContext, useEffect, createContext } from "react";

export const AuthContext = createContext({});

function useAuth() {
  const [token, setToken] = useState(window.localStorage.getItem("token"));

  useEffect(() => {
    if (token) {
      window.localStorage.setItem("token", token);
    } else {
      window.localStorage.removeItem("token");
    }
  }, [token]);

  return { token, setToken };
}

export const useAuthContext = () => useContext(AuthContext);

export { useAuth };
