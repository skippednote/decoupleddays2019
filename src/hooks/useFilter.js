import { useState, useEffect } from "react";
import { getDate } from "../utils/getDate";

export function useFilter(sessions) {
  const [filtered, setFiltered] = useState(sessions);
  const [filter, setFilter] = useState({
    category: "All",
    date: "All",
    input: ""
  });

  useEffect(() => {
    const filteredSession = sessions.filter(session => {
      const { date: sessionDate } = getDate(session.field_date);
      return (
        (filter.category === "All" ||
          session.field_category.name === filter.category) &&
        (filter.date === "All" || sessionDate === filter.date) &&
        (filter.input === "" || session.title.indexOf(filter.input) > -1)
      );
    });

    setFiltered(filteredSession);
  }, [sessions, filter]);

  return { filter, setFilter, filtered };
}
