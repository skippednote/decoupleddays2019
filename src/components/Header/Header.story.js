import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { MemoryRouter } from "react-router-dom";
import Header from "./";
import { AuthContext } from "../../hooks/useAuth";

storiesOf("Header", module)
  .add("default", () => (
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  ))
  .add("logged in", () => (
    <MemoryRouter>
      <AuthContext.Provider value={{ token: "yo", setToken: action() }}>
        <Header />
      </AuthContext.Provider>
    </MemoryRouter>
  ));
