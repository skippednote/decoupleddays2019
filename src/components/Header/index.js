import React from "react";
import { NavLink } from "react-router-dom";
import { FiHome, FiLogIn, FiLogOut, FiFilePlus } from "react-icons/fi";
import { useAuthContext } from "../../hooks/useAuth";
import Button from "../Button";

function Header() {
  const { token, setToken } = useAuthContext();

  return (
    <header>
      <ul className="navigation">
        <li>
          <NavLink exact to="/" activeclassname="active" className="button">
            <FiHome /> <span>Home</span>
          </NavLink>
        </li>

        {token && (
          <>
            <li>
              <NavLink
                exact
                to="/create"
                activeclassname="active"
                className="button"
              >
                <FiFilePlus /> <span>Create a session</span>
              </NavLink>
            </li>
            <li>
              <Button
                className="button"
                onClick={() => setToken(null)}
                activeclassname="active"
              >
                <FiLogOut /> <span>Logout</span>
              </Button>
            </li>
          </>
        )}

        {!token && (
          <li>
            <NavLink
              exact
              to="/login"
              activeclassname="active"
              className="button"
            >
              <FiLogIn /> <span>Login</span>
            </NavLink>
          </li>
        )}
      </ul>
    </header>
  );
}

export default Header;
