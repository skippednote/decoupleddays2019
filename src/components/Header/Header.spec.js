import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "./";
import { AuthContext } from "../../hooks/useAuth";

it("renders the component", () => {
  const { container } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});

it("renders the logout button when logged in", () => {
  const { queryByText } = render(
    <MemoryRouter>
      <AuthContext.Provider value={{ token: "yo" }}>
        <Header />
      </AuthContext.Provider>
    </MemoryRouter>
  );
  expect(queryByText("Logout")).toBeTruthy();
});

it("logs out the user on clicking logout button", () => {
  const setTokenMock = jest.fn();
  const { getByText } = render(
    <MemoryRouter>
      <AuthContext.Provider value={{ token: "yo", setToken: setTokenMock }}>
        <Header />
      </AuthContext.Provider>
    </MemoryRouter>
  );
  fireEvent.click(getByText("Logout"));
  expect(setTokenMock).toHaveBeenCalled();
  expect(setTokenMock).toHaveBeenCalledWith(null);
});
