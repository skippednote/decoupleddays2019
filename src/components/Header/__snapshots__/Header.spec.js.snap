// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div>
  <header>
    <ul
      class="navigation"
    >
      <li>
        <a
          activeclassname="active"
          aria-current="page"
          class="button active"
          href="/"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
            />
            <polyline
              points="9 22 9 12 15 12 15 22"
            />
          </svg>
           
          <span>
            Home
          </span>
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          class="button"
          href="/login"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
            />
            <polyline
              points="10 17 15 12 10 7"
            />
            <line
              x1="15"
              x2="3"
              y1="12"
              y2="12"
            />
          </svg>
           
          <span>
            Login
          </span>
        </a>
      </li>
    </ul>
  </header>
</div>
`;
